### ISSUE TITLE
GDScript 2.0: "Cannot convert argument 1 from Object to Object" when signal parameter is typed with an inner class.

### ISSUE THREAD

**Issue by nathanfranke** [godotengine/godot#62982]
### Godot version

v4.0.alpha.custom_build [8b059d47f]

### System information

Arch on 5.18.10-arch1-1

### Issue description

Looks similar to https://github.com/godotengine/godot/issues/26047

I think `game: Global.Game` is the problem. Maybe it thinks the type is Global?

```
E 0:00:04:0641   _ready: Error calling from signal 'game_changed' to callable: 'Node(scene.gd)::_on_game_changed': Cannot convert argument 1 from Object to Object..
  <C++ Source>   core/object/object.cpp:1025 @ emit_signalp()
  <Stack Trace>  global.gd:10 @ _ready()
```

### Steps to reproduce

`global.gd` in autoload `Global`

```gdscript
extends Node

signal game_changed(game)

class Game:
	extends Node

func _ready() -> void:
	await get_tree().process_frame
	game_changed.emit(Game.new())

```

`scene.gd`

```gdscript
extends Node

func _ready() -> void:
	Global.game_changed.connect(_on_game_changed)

func _on_game_changed(game: Global.Game) -> void:
	print(game)

```

Cannot reproduce if everything is in one file

### Minimal reproduction project

[test10.zip](https://github.com/godotengine/godot/files/9104420/test10.zip)

**Comment by Erveon** [comment]
I have the same issue when creating an event bus that uses inner classes as their parameter.

Global Events.gd
```gdscript
extends Node

class EntityDamagedMessage extends Object:
	var entity: Node2D
	var damager: Node2D
	var amount: float

signal entity_damaged(message: EntityDamagedMessage)
```

Another script
```gdscript
var message := Events.EntityDamagedMessage.new()
Events.emit_signal("entity_damaged", message)
```

Are there any known workarounds?

**Comment by rune-scape** [comment]
@akien-mga just tested that this is fixed by #68374. A better error message here would have revealed that it was probably trying to convert from `Global.Game` to `Global`.
